### PR TITLE
refactor to use marshaller and pubkey converter from config

### PR DESCRIPTION
**Pull request by ssd04**
Refactor to initiate core components from config:
- marshaller
- pubkey converter

### CODE REVIEW

File: api/groups/eventsData.go [hunk 1]
```diff
@@ -23,14 +23,14 @@ func UnmarshallBlockDataV1(marshalledData []byte) (*data.SaveBlockData, error) {
...
-func UnmarshallBlockData(marshalledData []byte) (*data.ArgsSaveBlockData, error) {
+func UnmarshallBlockData(marshaller marshal.Marshalizer, marshalledData []byte) (*data.ArgsSaveBlockData, error) {
 	var argsBlockS *outport.OutportBlock
 	err := json.Unmarshal(marshalledData, &argsBlockS)
 	if err != nil {
 		return nil, err
 	}
 
-	header, err := unmarshal.GetHeaderFromBytes(&marshal.JsonMarshalizer{}, core.HeaderType(argsBlockS.BlockData.HeaderType), argsBlockS.BlockData.HeaderBytes)
+	header, err := unmarshal.GetHeaderFromBytes(marshaller, core.HeaderType(argsBlockS.BlockData.HeaderType), argsBlockS.BlockData.HeaderBytes)
```
Comment (mariusmihaic) [inline api/groups/eventsData.go:33]: Just a heads up here.
This func will be deleted and replaced with a factory.
Not sure if you should integrate the new one here or in a future PR. Here are the changes: https://github.com/multiversx/mx-chain-core-go/pull/190

File: config/config.go [hunk 2]
```diff
@@ -55,8 +74,8 @@ type FlagsConfig struct {
 }
 
 // LoadConfig return a GeneralConfig instance by reading the provided toml file
```
Comment (mariusmihaic) [inline config/config.go:76]:
```suggestion
// LoadConfig return a Config instance by reading the provided toml file
```

File: factory/processFactory.go [hunk 3]
```diff
@@ -70,3 +70,12 @@ func CreateEventsInterceptor() (process.EventsInterceptor, error) {
 
 	return process.NewEventsInterceptor(argsEventsInterceptor)
 }
+
+func getPubKeyConverter(cfg *config.GeneralConfig) (core.PubkeyConverter, error) {
```
Comment (mariusmihaic) [inline factory/processFactory.go:74]: Why don't you use this func in `CreateEventsInterceptor` when creating `pubKeyConverter` ?

File: mocks/marshalizerMock.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package mocks
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+// ErrMockMarshalizer -
+var ErrMockMarshalizer = errors.New("MarshalizerMock generic error")
```
Comment (mariusmihaic) [inline mocks/marshalizerMock.go:9]: No need for this to be exported

File: mocks/marshalizerMock.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+// MarshalizerMock that will be used for testing
+type MarshalizerMock struct {
```
Comment (mariusmihaic) [inline mocks/marshalizerMock.go:12]: This mock keeps being copied in every project (I also had to copy it in sovereign notifier).
We should really move it to core (not necessarily here in this PR)

File: api/groups/eventsData.go (hunk 1 above)
Comment (ssd04) [reply api/groups/eventsData.go:33]: yes, i would do it in a next PR

File: factory/processFactory.go (hunk 3 above)
Comment (ssd04) [reply factory/processFactory.go:74]: ahh, forgot to replace, replaced now

File: mocks/marshalizerMock.go (hunk 5 above)
Comment (ssd04) [reply mocks/marshalizerMock.go]: thanks for the heads up, actually there is already a mock implementation in core, refactored to use it from there

**Review by sstanculeanu — DISMISSED** [review]

File: config/config.go [hunk 6]
```diff
@@ -73,7 +73,7 @@ type FlagsConfig struct {
 	APIType           string
 }
 
-// LoadConfig return a GeneralConfig instance by reading the provided toml file
+// LoadConfig return a General instance by reading the provided toml file
```
Comment (mariusmihaic) [inline config/config.go:76]:
```suggestion
// LoadConfig return a Config instance by reading the provided toml file
```

**Review by mariusmihaic — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**